sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -1697,11 +1696,17 @@ public PCollection<KafkaRecord<K, V>> expand(PBegin input) {
         }
 
         if (kafkaRead.isRedistributed()) {
-          // fail here instead.
-          checkArgument(
-              kafkaRead.isCommitOffsetsInFinalizeEnabled(),
-              "commitOffsetsInFinalize() can't be enabled with isRedistributed");
+          if (kafkaRead.isCommitOffsetsInFinalizeEnabled()) {
+            LOG.warn(
+                "commitOffsetsInFinalize() will not capture all work processed if set with withRedistribute()");
+          }
+          if (Boolean.TRUE.equals(
+              kafkaRead.getConsumerConfig().get(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG))) {
+            LOG.warn(
+                "config.ENABLE_AUTO_COMMIT_CONFIG doesn't need to be set with withRedistribute()");
```

Comment:
It doesn't need to but it seems like it could still be desirable if they are using the offsets for monitoring. 

Should we just remove this log? It seems the correctness is the same as if we were shuffling persistently though the window is perhaps larger if a drain is being performed.

Reply:
Ack, removed it.